wp-content/themes/pub/wporg-learn-2024/inc/block-config.php

```diff
@@ -557,3 +559,78 @@ function inject_other_filters( $key ) {
 		printf( '<input type="hidden" name="s" value="%s" />', esc_attr( $wp_query->query['s'] ) );
 	}
 }
+
+/**
+ * Filter the block data for the student course list.
+ *
+ * @param array $parsed_block The parsed block data.
+ * @return array The updated block data.
+ */
+function filter_student_course_list( $parsed_block ) {
+	if (
+		'core/query' !== $parsed_block['blockName'] ||
+		'course' !== ( $parsed_block['attrs']['query']['postType'] ?? '' ) ||
+		! isset( $parsed_block['attrs']['queryId'] ) ||
+		! is_page( 'my-courses' )
```

Comment:
The My Courses page can be changed via Sensei settings, which is why we have this function in the plugin: https://github.com/WordPress/Learn/blob/e851a022edd129f7bb8d1c8780729e64ca31bf89/wp-content/plugins/wporg-learn/inc/sensei.php#L262

Maybe you could check if the current page id matches `Sensei()->settings->get_my_courses_page_id()`?